lib/segment/src/index/field_index/full_text_index/postings_iterator.rs

```diff
@@ -14,7 +15,7 @@ pub fn intersect_postings_iterator<'a>(
     let smallest_posting = postings.remove(smallest_posting_idx);
 
     let and_iter = smallest_posting
-        .iter()
+        .iter(filter)
         .filter(move |doc_id| postings.iter().all(|posting| posting.contains(doc_id)));
```

Comment:
I didn't add deleted condition in `filter` because we calculate the set intersection here. And all deleted id's are filtered by `iter(filter)`